.github/workflows/ci.yml

```diff
@@ -0,0 +1,88 @@
+name: CI & Deployment
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+  release:
+    types: [published]
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  id-token: write
+  contents: write
+
+env:
+  TSYS_REFERRER: ${{ vars.TSYS_REFERRER }}
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    name: Test and Build
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 7
+      - name: 📦 Install Dependencies
+        run: yarn install --immutable --immutable-cache
+      - name: 💨 ESLint
+        run: yarn lint
+      - name: 🧪 Tests
+        run: yarn test
+      - name: ✅ Build
+        run: yarn build
+      - name: Codecov
+        uses: codecov/codecov-action@v3
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          fail_ci_if_error: true
+  deploy:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'release'
+    name: Deploy to NPM
+    needs: [test]
+    steps:
+      - uses: actions/checkout@v3
+      # Setup .npmrc file to publish to npm
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 7
+          registry-url: "https://registry.npmjs.org"
+          scope: "CruPCI"
+      - run: yarn install --immutable --immutable-cache
+      - run: yarn publish
```

Comment:
The Travis deploy process kept the tags, which drove the version number. How is that being captured here? Do we need the `--new-version` or `--tag` arguments?

Reply:
I assume you're referring to the `tags: true` on .travis.yml. This means that it will only deploy to npm when a new version of your package is cut. https://docs.travis-ci.com/user/deployment/npm/#what-to-release.

I believe the version number is determined by the version number in package.json

Reply:
I think the process was to merge the branch to master, then create a release in GitHub with a tag specifying the version number. This release creation would then trigger a release to npm.

Reply:
On the publishing of a GH release, the package does get deployed to NPM.
We can make it so the `tag` is what sets the version instead of the package.json, by doing this:
```yarn publish --new-version ${{ github.event.release.tag_name }}```